packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -75,8 +74,18 @@ export class BaseInvocationScope<TReturn = any> {
    * @returns An array of contract calls.
    */
   protected get calls() {
+    const script = getContractCallScript(this.functionInvocationScopes);
+    const provider = this.program.provider as Provider;
```

Comment:
Why is this `as Provider` cast necessary?

Reply:
@arboleya this cast is necessary because the `AbstractProgram` class has its own custom definition of the `provider` field, which only includes the `sendTransaction` method and nothing else:

https://github.com/FuelLabs/fuels-ts/blob/5310e8b5226a74058b85a6f1ca3671b6b5b5c5b7/packages/interfaces/src/index.ts#L63-L66

I believe this might be to avoid cyclic dependencies with the `providers` package.

Reply:
I see; maybe an `AbstractProvider` could be added to the `interfaces` package to solve this?

> I'm not a fan of using abstract classes. TBH, I'm against _[the vast majority]_ of abstractions, but it is what it is. I reckon the `interfaces` package only exists for circumventing cyclic dependencies — something to consider.

Reply:
@arboleya Fair, I'll file a follow-up for this too and fix it in a separate PR

Reply:
- https://github.com/FuelLabs/fuels-ts/issues/1257